### PR TITLE
Add vmin vmax args to ``plot``

### PR DESCRIPTION
**Pull request by Tmodrzyk**
This PR implements the "absolute scale" feature of #927. When selecting the ``clip`` rescale mode, you can pass ``vmin`` and ``vmax`` in ``**kwargs_imshow`` and it will be used to clip the signals and plot the images.
It can be used as follows:
```python
import deepinv as dinv

x1 = dinv.utils.load_example("butterfly.png", img_size=(128, 128), grayscale=True)
physics1 = dinv.physics.PoissonNoise(gain=1 / 5)
physics2 = dinv.physics.PoissonNoise(gain=1 / 100)

y1 = physics1(x1)
y2 = physics2(x1)
vmin = min(x1.min().item(), y1.min().item(), y2.min().item())
vmax = max(x1.max().item(), y1.max().item(), y2.max().item())
dinv.utils.plot(
    [x1, y1, y2],
    titles=["Ground-truth image", "Noisy measurement 1", "Noisy measurement 2"],
    figsize=(10, 6),
    rescale_mode="clip",
    vmin=vmin,
    vmax=vmax,
)
```

### Checks to be done before submitting your PR

- [x] `python3 -m pytest deepinv/tests` runs successfully.
- [x] `black .` runs successfully.
- [ ] `make html` runs successfully (in the `docs/` directory).
- [x] Updated docstrings related to the changes (as applicable).
- [x] Added an entry to the [changelog.rst](https://github.com/deepinv/deepinv/blob/main/docs/source/changelog.rst).

### CODE REVIEW

File: deepinv/utils/plotting.py [hunk 1]
```diff
@@ -189,7 +189,7 @@ def prepare_images(x=None, y=None, x_net=None, x_nl=None, rescale_mode="min_max"
 
 
 @torch.no_grad()
-def preprocess_img(im, rescale_mode="min_max"):
+def preprocess_img(im, rescale_mode="min_max", vmin=None, vmax=None):
```
Comment (jscanvic) [inline deepinv/utils/plotting.py]:
```suggestion
def preprocess_img(im, rescale_mode="min_max", *, vmin: float | None = None, vmax: float | None = None):
```

File: deepinv/utils/signals.py [hunk 2]
```diff
@@ -44,7 +55,11 @@ def normalize_signal(inp, *, mode):
         inp[indices] = inp[indices].clamp(min=0.0, max=1.0)
     elif mode == "clip":
         # Clamp every batched signal between zero and one
-        inp = inp.clamp(min=0.0, max=1.0)
+        if vmin is None:
+            vmin = 0.0
+        if vmax is None:
+            vmax = 1.0
+        inp = inp.clamp(min=vmin, max=vmax)
```
Comment (jscanvic) [inline deepinv/utils/signals.py]: You also need to rescale `inp` so the [vmin, vmax] range maps to [0, 1] don't you?

**Review by jscanvic — CHANGES_REQUESTED** [review]
Thanks Thibaut! Some comments

* Can you check if matplotlib supports the use of vmin/vmax in imshow for RGB images? (I think not, we might need to preprocess the image ourselves here)
* It's a bit awkward that `normalize_signal` returns a signal that's not normalized between zero and one despite being the purpose/description of the function
* Update the docstrings
* Can you add a warning for the use of incompatible parameter values? (e.g., vmin and vmax being set while rescale_mode is not set to "clip")
* Update the changelog

**Comment by Tmodrzyk** [comment]
Hi @jscanvic, I took into account your comments. However for:
> _it's a bit awkward that normalize_signal returns a signal that's not normalized between zero and one despite being the purpose/description of the function_

I'm not sure if we should rename the function ``normalize_signal`` to something else, or remove the clipping option and put it in ``dinv.utils.preprocess_img``. The function ``normalize_signal`` (with ``clip`` rescale mode) is already used elsewhere in the repo, but I can fix it manually.

File: deepinv/utils/plotting.py [hunk 3]
```diff
@@ -214,9 +218,19 @@ def preprocess_img(im, rescale_mode="min_max", *, vmin: float | None = None, vma
...
+                "The vmin and vmax arguments are ignored when using 'min_max' rescaling.",
+                UserWarning,
+                stacklevel=2,
+            )
+        if vmin is not None and vmax is not None and vmin >= vmax:
+            raise ValueError(
+                f"vmin should be strictly less than vmax, got vmin={vmin} and vmax={vmax}."
+            )
+        # Normalize values between zero and one
+        im = normalize_signal(im, mode=rescale_mode, vmin=vmin, vmax=vmax)
```
Comment (jscanvic) [inline deepinv/utils/plotting.py]: It should be outside the if right?

Comment (Tmodrzyk) [reply deepinv/utils/plotting.py]: Oops, thank you !

**Comment by jscanvic** [comment]
Ty for making the change!

RGB images don't seem to work afaict. If I'm not mistaken matplotlib internally overwrites vmin and vmax to 0 and 1 for them. A possible solution is to do the rescaling+clipping ourselves before passing the image to imshow (without passing vmin and vmax to imshow)

**Reproducing**

```python
import deepinv as dinv

x1 = dinv.utils.load_example("butterfly.png", img_size=(128, 128), grayscale=False)
x2 = x1 + 100
x3 = x1 + 100.5
x4 = x1 + 200

dinv.utils.plot(
    [x1, x2, x3, x4],
    rescale_mode="clip",
    vmin=100,
    vmax=101,
    cbar=True,
)
```

**Expected behavior**

The 1st image should be black (clipped out), the 2nd image should show the butterfly normally, the 3rd image should show a out-of-tone butterfly with clipped pixels, the 4th image should be white (clipped out)

**Observed behavior**

Every image is white

**Comment by Tmodrzyk** [comment]
Hi @jscanvic thanks for the test, the one I conducted was not very telling for RGB. Now it should have the expected behavior. 
Basically now in ``preprocess_img`` after calling ``normlize_signal`` to clip the images, I rescale to [0, 1]. ``vmin`` and ``vmax`` are reset to 0 and 1 before being passed to ``imshow``. 
I hope this won't cause some unexpected behaviors down the line...

**Comment by jscanvic** [comment]
Ty! Will have a closer look shortly

File: deepinv/utils/plotting.py [hunk 4]
```diff
@@ -214,10 +218,23 @@ def preprocess_img(im, rescale_mode="min_max"):
     # Cast image values to float32 numbers
     # NOTE: Why is it needed?
     im = im.type(torch.float32)
+    if rescale_mode == "min_max":
```
Comment (jscanvic) [inline deepinv/utils/plotting.py]: * I'd move that to `normalize_signal`
* I think it makes more sense to test if `rescale_mode != "clip"` instead of testing that it's `min_max`, it's more that vmin/vmax is only compatible with clip than it's not compatible with min_max

File: deepinv/utils/plotting.py [hunk 5]
```diff
@@ -214,10 +218,23 @@ def preprocess_img(im, rescale_mode="min_max"):
...
+        raise ValueError(
+            f"vmin should be strictly less than vmax, got vmin={vmin} and vmax={vmax}."
+        )
+    # Normalize signal between 0 and 1
+    im = normalize_signal(im, mode=rescale_mode, vmin=vmin, vmax=vmax)
 
-    # Normalize values between zero and one
-    im = normalize_signal(im, mode=rescale_mode)
-
+    if rescale_mode == "clip":
```
Comment (jscanvic) [inline deepinv/utils/plotting.py]: I'd move that to `normalize_signal` too

That way normalize_signal always gives out images between 0 and 1

File: deepinv/utils/plotting.py [hunk 6]
```diff
@@ -205,6 +207,8 @@ def preprocess_img(im, rescale_mode="min_max"):
 
     :param torch.Tensor im: the batch of images to preprocess, it is expected to be of shape (B, C, *).
     :param str rescale_mode: the normalization mode, either 'min_max' or 'clip'.
+    ;param float vmin: minimum value for clipping when using 'clip' rescaling.
```
Comment (jscanvic) [inline deepinv/utils/plotting.py]:
```suggestion
    :param float, None vmin: minimum value for clipping when using 'clip' rescaling.
```

File: deepinv/utils/plotting.py [hunk 7]
```diff
@@ -205,6 +207,8 @@ def preprocess_img(im, rescale_mode="min_max"):
...
+    ;param float vmax: maximum value for clipping when using 'clip' rescaling.
```
Comment (jscanvic) [inline deepinv/utils/plotting.py]:
```suggestion
    :param float, None vmax: maximum value for clipping when using 'clip' rescaling.
```

File: deepinv/utils/plotting.py [hunk 8]
```diff
@@ -214,10 +218,23 @@ def preprocess_img(im, rescale_mode="min_max"):
     # Cast image values to float32 numbers
     # NOTE: Why is it needed?
     im = im.type(torch.float32)
+    if rescale_mode == "min_max":
+        if vmin is not None or vmax is not None:
+            warn(
+                "The vmin and vmax arguments are ignored when using 'min_max' rescaling.",
```
Comment (jscanvic) [inline deepinv/utils/plotting.py]: Update the message too if you invert the condition

File: deepinv/utils/plotting.py [hunk 9]
```diff
@@ -353,14 +370,20 @@ def plot(
...
-            col_imgs.append(
+            row_imgs.append(
                 im[i, ...].detach().permute(1, 2, 0).squeeze().cpu().numpy()
             )
-        imgs.append(col_imgs)
+        imgs.append(row_imgs)
+
+    imshow_kwargs = dict(imshow_kwargs)
+    imshow_kwargs.setdefault("vmin", 0)
+    imshow_kwargs.setdefault("vmax", 1)
```
Comment (jscanvic) [inline deepinv/utils/plotting.py]: No need for that if you put vmin and vmax as regular parameters to the function instead of having them in the imshow_kwargs in the first place

**Comment by Tmodrzyk** [comment]
I implemented your recommendations, I'll let you know once the CI passes. :)

**Comment by Tmodrzyk** [comment]
I think this is good to go @jscanvic :smile:

File: deepinv/utils/signals.py [hunk 10]
```diff
@@ -25,6 +33,11 @@ def normalize_signal(inp, *, mode):
         # Clone the signal to avoid input mutations
         inp = inp.clone()
 
+        if vmin is not None:
+            minimum_intensity = torch.full_like(minimum_intensity, vmin)
+        if vmax is not None:
+            maximum_intensity = torch.full_like(maximum_intensity, vmax)
+
         # The indices corresponding to the non-constant batched signals
```
Comment (jscanvic) [inline deepinv/utils/signals.py]:
```suggestion
        # The indices corresponding to the non-constant batched signals
```

I think we should leave the minmax rescaling mode as is

File: deepinv/utils/plotting.py [hunk 11]
```diff
@@ -214,9 +218,19 @@ def preprocess_img(im, rescale_mode="min_max"):
...
+        if vmin is not None or vmax is not None:
+            warn(
+                "The vmin and vmax arguments are used only when using 'clip' rescaling.",
+                UserWarning,
+                stacklevel=2,
+            )
+    if vmin is not None and vmax is not None and vmin >= vmax:
+        raise ValueError(
+            f"vmin should be strictly less than vmax, got vmin={vmin} and vmax={vmax}."
+        )
```
Comment (jscanvic) [inline deepinv/utils/plotting.py]: Please put this whole block at the beginning of `normalize_signal`

Comment (Tmodrzyk) [reply deepinv/utils/plotting.py]: Done in [afc3635](https://github.com/deepinv/deepinv/pull/967/commits/afc3635465bb408003306b69f0ecdcf34b383ef2) !

**Review by jscanvic — APPROVED** [review]
LGTM thanks Thibaut!
